internal/commands/app/create.go

```diff
@@ -122,7 +151,12 @@ func (cmd *CommandCreate) Handler(profile *user.Profile, ui terminal.UI, clients
 	appRealm, err := clients.Realm.CreateApp(
 		groupID,
 		cmd.inputs.Name,
-		realm.AppMeta{cmd.inputs.Location, cmd.inputs.DeploymentModel, cmd.inputs.Environment},
+		realm.AppMeta{
+			cmd.inputs.Location,
+			cmd.inputs.DeploymentModel,
+			cmd.inputs.Environment,
+			templateID,
```

Comment:
I can't comment on the line exactly [L#146](https://github.com/10gen/realm-cli/pull/163/files#diff-56dff07c87480c9d173d72d506ffac3e9e906a3ff89db6be5d819a1143655c63R146), but the `cmd.display` function relies on `inputs` having the values desired to be played back to the user (if run with `--dry-run`)

for that reason, I think it may be desirable to mutate the value of `cmd.inputs.Template` rather than return `templateID` from the `inputs.resolveTemplateID` function.  it's actually a very minor use-case/unlikely scenario here, and the command printed out here would still "work" (just assuming the user resolves the template id the same way if through prompts).